js-exercises/all-promises/allPromises.js

```diff
@@ -0,0 +1,18 @@
+const allPromises = args => {
+  if(args===undefined){
+    return Promise.resolve()
+  }
+  let arr=[];
+  args.map((value,index)=>{
+    if(value instanceof Promise){
+      value.then(x=>{
+       arr[index]=x
+      })
+    }else{
+      arr[index]=value
+    }
+  })
+  return Promise.resolve(arr)
+};
+
+export { allPromises };
```

Comment:
Hey, good use of declarative looping. However, looking at the map method, gives me an impression that we are returning an array (and doing something with it) but that is not the case, so I think forEach method could be used.
Apart from that, you could have kept track of all the promises to make sure that they are resolved, so that you can then resolve the main (outer) promise. Because right now, I think this code will not work for asynchronous operations.
So, on line 15, instead of immediately resolving the promise, make sure to resolve it only after all the promises have been resolved. And also try to consider the rejection of promises.
Other than that, the code is clean and easy to understand. :100: :zap: